tests/backup_helper.go

```diff
@@ -8540,3 +8542,40 @@ func GetUpdatedKubeVirtVMSpecForBackup(scheduledAppContextsToBackup []*scheduler
 	}
 	return nil
 }
+
+// GetPVCListForNamespace retrieves the list of PVCs in the specified namespace.
+func GetPVCListForNamespace(namespace string) ([]string, error) {
+	k8sCore := core.Instance()
+	pvcList, err := k8sCore.GetPersistentVolumeClaims(namespace, make(map[string]string))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get PVCs in namespace %s: %w", namespace, err)
+	}
+	// Extract PVC names from the list
+	var pvcNameList []string
+	for _, pvc := range pvcList.Items {
+		pvcNameList = append(pvcNameList, pvc.Name)
+	}
+	return pvcNameList, nil
+}
+
+// ValidatePVCCleanup checks if there is a mismatch between the original PVC list and the current one.
+func ValidatePVCCleanup(pvcListBefore, pvcListAfter []string) error {
+	if len(pvcListBefore) != len(pvcListAfter) {
+		fmt.Println("Expected PVCs:", strings.Join(pvcListBefore, ", "))
+		fmt.Println("Actual PVCs:", strings.Join(pvcListAfter, ", "))
+		return fmt.Errorf("mismatch in PVC count")
+	}
+	var mismatchedPVCs []string
+	for i := 0; i < len(pvcListBefore); i++ {
+		if pvcListBefore[i] != pvcListAfter[i] {
+			mismatchedPVCs = append(mismatchedPVCs, pvcListBefore[i])
+		}
```

Comment:
This logic may not work always. What if the size of `pvcListAfter` is less than that of `pvcListBefore`
Also, it is better to print it in a way which say `pvc1 is present in pvcListBefore but not in pvcListAfter`

Look at this function and implement it accordingly - https://go.dev/play/p/V7n5b7d7HpN

Reply:
@mkoppal-px If the the size of `pvcListAfter `is less than that of `pvcListBefore`, the same logic would work;
We would get the error of mismatch in PVC count with both the PVC lists.
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/5042/console

I will modify the function to print only the mismatched PVC.

Reply:
@mkoppal-px I have modified the logic accordingly and have updated the run details for all scenarios in the description. I didn't run for all testcases but instead just for BasicBackupCreation. Please advise.